pkg/plugin/oauth2/setup.go

```diff
@@ -19,14 +32,79 @@ type Config struct {
 	ServerName string `json:"server_name"`
 }
 
-func setupOAuth2(route *proxy.Route, p plugin.Params) error {
+func onAdminAPIStartup(event interface{}) error {
+	e, ok := event.(plugin.OnAdminAPIStartup)
+	if !ok {
+		return errors.New("Could not convert event to admin startup type")
+	}
+
+	adminRouter = e.Router
```

Comment:
and here we could have `p.adminRouter = e.Router` assigning the variable to the struct